tests/test_layercatalog.py

```diff
@@ -231,6 +234,158 @@ def test_get_layer_catalog_opensearch_enrich_creodias(requests_mock, vault):
     ]
 
 
+@pytest.mark.skip("Run manually when changing layercatalog.json files")
+def test_layer_catalog_step_resolution(vault):
+
+    # TODO: Move to integrationtests
+
+    with mock.patch("openeogeotrellis.layercatalog.ConfigParams") as ConfigParams:
+        ConfigParams.return_value.layer_catalog_metadata_files = [
+            "/home/emile/openeo/openeo-deploy/mep/layercatalog.json",
+            # "/home/emile/VITO/enriched_layercatalog.json",
+            # "/home/emile/openeo/openeo-geotrellis-kubernetes/docker/creo_layercatalog.json",
+        ]
+        catalog = get_layer_catalog(vault, opensearch_enrich=True)
+        all_metadata = catalog.get_all_metadata()
+
+    warnings = ""
+    for layer in all_metadata:
+        # if layer["id"] in ["MAPEO_WATER_TUR_V1", "SENTINEL1_GRD_SIGMA0", "S1_GRD_SIGMA0_ASCENDING", "S1_GRD_SIGMA0_DESCENDING"]:
+        #     continue
+        metadata = GeopysparkCubeMetadata(catalog.get_collection_metadata(collection_id=layer["id"]))
+        print(f"\n{layer['id']=}")
+
+        temporal_step_str = metadata.get("cube:dimensions", "t", "step", default=None)
+        if temporal_step_str:
+            print(f"{temporal_step_str=}")
+
+        if metadata.get("id") == 'WATER_BODIES':
+            print("ha")
+        gsd_in_meter = metadata.get_GSD_in_meters()
+        print(f"{gsd_in_meter=}")
+        datasource_type = metadata.get("_vito", "data_source", "type")
+        warn_str = layer["id"] + " datasource_type:" + str(datasource_type) + "\n"
+
+        def clean_2D_tuple(tuple_to_clean):
```

Comment:
is it intentional to duplicate this clean_2D_tuple from geopysparkcubemetadata.py?

Reply:
no, deduped. Thanks!